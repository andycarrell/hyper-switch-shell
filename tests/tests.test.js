const { middleware } = require("../index");

global.Date = {
  now: () => 1524965450320
};

test("any given action, should be passed on with next", () => {
  const mockNext = jest.fn();
  const mockAction = {
    type: "TESTING_PASS_THROUGH"
  };
  middleware({ dispatch: () => {} })(mockNext)(mockAction);

  expect(mockNext).toMatchSnapshot();
});

test("action type CONFIG_LOAD should not call dispatch", () => {
  const mockDispatch = jest.fn();
  const testFn = middleware({ dispatch: mockDispatch })(() => {});

  const mockAction = { type: "CONFIG_LOAD" };
  testFn(mockAction);

  expect(mockDispatch).not.toBeCalled();
});

test("action type CONFIG_LOAD should not call dispatch", () => {
  const mockDispatch = jest.fn();
  const testFn = middleware({ dispatch: mockDispatch })(() => {});

  const mockAction = { type: "CONFIG_RELOAD" };
  testFn(mockAction);

  expect(mockDispatch).not.toBeCalled();
});

test("action type UI_COMMAND_EXEC should not call dispatch, if there is no command", () => {
  const mockDispatch = jest.fn();
  const testFn = middleware({ dispatch: mockDispatch })(() => {});

  const mockAction = { type: "UI_COMMAND_EXEC" };

  testFn(mockAction);

  expect(mockDispatch).not.toBeCalled();
});

test("action type UI_COMMAND_EXEC should call dispatch, when command matches", () => {
  const mockDispatch = jest.fn();
  const testFn = middleware({ dispatch: mockDispatch })(() => {});

  const testKey = "default";

  // setup config
  const mockConfigAction = {
    type: "CONFIG_LOAD",
    config: {
      shells: {
        [testKey]: {
          shell: "bash",
          shellArgs: ["shell arg"]
        }
      }
    }
  };

  testFn(mockConfigAction);

  const mockAction = {
    type: "UI_COMMAND_EXEC",
    command: `shell:load:${testKey}`
  };

  testFn(mockAction);

  expect(mockDispatch).toMatchSnapshot();
});
