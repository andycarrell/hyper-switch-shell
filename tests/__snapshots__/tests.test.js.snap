// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action type UI_COMMAND_EXEC should call dispatch, when command matches 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "config": Object {
          "shell": "bash",
          "shellArgs": Array [
            "shell arg",
          ],
          "shells": Object {
            "default": Object {
              "shell": "bash",
              "shellArgs": Array [
                "shell arg",
              ],
            },
          },
        },
        "now": 1524965450320,
        "pluginName": "hyper-switch-shell",
        "type": "CONFIG_RELOAD",
      },
    ],
  ],
}
`;

exports[`any given action, should be passed on with next 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "TESTING_PASS_THROUGH",
      },
    ],
  ],
}
`;
